### PR TITLE
adding text for lr_find, early_stopping, checkpoint, predict_video, publishing deep learning package

### PR DESCRIPTION
**Pull request by AngadKundra**
…ublishing dlpk

\<insert pull request description here\>
LR Find
Early stopping
Checkpoint
DLPK publishing
Predict Video
-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [x] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports?
- [x] All `GIS` object instantiations are one of the following?
    - `gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')`
    - `gis = GIS(profile="your_online_profile")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal', 'arcgis_python', 'amazing_arcgis_123')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [x] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [x] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [x] Code refactored & split out across multiple cells, useful comments?
- [x] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [x] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [x] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [x] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@AngadKundra Could you please link this to the issue created?

**Comment by AngadKundra** [comment]
https://github.com/ArcGIS/geosaurus/issues/2830

**Comment by priyankatuteja** [comment]
@moonlanderr Could you please review this?

File: guide/14-deep-learning/object-detection.ipynb [hunk 1]
```diff
@@ -69,7 +69,7 @@
        "                        </a>\n",
```
Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <ol><li>Spellcheck</li></ol><ul><li><em>given the input imagery, </em><strong><em>along with labelled</em></strong><em> vector data or classified images.&nbsp;</em></li><li><em>learning training samples are small </em><strong><em>sub images</em></strong><em>, called image chips, and contain the feature</em></li><li>package as well as by third-party deep learning <strong>libraries</strong>, such as TensorFlow</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <p>rephrase</p><blockquote>Find the <strong>efficient</strong> learning rate</blockquote><blockquote><br></blockquote><blockquote><strong>Choosing</strong> a very small learning rate leads to very slow training of the model</blockquote><blockquote><br></blockquote><blockquote>while an extremely high rate can 'overshoot' the minima where the loss (or error rate) is lowest, and prevent the model from converging</blockquote><blockquote><br></blockquote><blockquote><code>arcgis.learn</code>&nbsp;includes fast.ai's learning rate finder, which is accessible through the model's&nbsp;<code>lr_find()</code>&nbsp;method, that can automatically select an optimum learning rate, without requiring repeated experiments.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <p>This graph does not seem to be an output from the code ssd.lr_find(). This looks like an image added using markdown cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <p>spellcheck</p><ul><li>As <strong>discussed</strong>&nbsp;earlier</li><li>for fine-tuning the <strong>earlier</strong> layers is usually one tenth&nbsp;</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <p>Title could be changed as --  <strong>Model Deployment</strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <ol><li>paragraph could be started by this line --</li></ol><blockquote>Now, we can further deploy the trained model to run prediction on a video. <code>predict_video</code>&nbsp;function is used to run prediction on a video and</blockquote><p> </p><p> </p><p><em>You can select assignment threshold for the assignment of trackers, the number of frames the detection should persist if object is not found and threshold of number of frames to start detecting an object. These values can be passed as a dictionary in the argument tracking options.</em></p><p> </p><p>this above could be elaborated a bit and rephrased as : -</p><p> </p><blockquote>The following options/parameters can be selected in the predict video function by the user : -</blockquote><p> </p><blockquote>Assignment threshold <strong>- </strong> There might be multiple trackers detecting and tracking objects. The Intersection over Union (iou) threshold can be set to assign a tracker with the mentioned threshold value.</blockquote><blockquote>Then the number of frames the object remains absent from the frame to be considered as vanished.</blockquote><blockquote>Also the number of frames an object remains present in the frame to start tracking it.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

File: guide/14-deep-learning/object-detection.ipynb [hunk 2]
```diff
@@ -6,7 +6,7 @@
    "source": [
```
Comment (moonlanderr) [inline guide/14-deep-learning/object-detection.ipynb]: <p>in the first line of the paragraph it would be informative if the word VMTI (<em>output VMTI predictions</em> ) is elaborated  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (AtmaMani) [inline guide/14-deep-learning/object-detection.ipynb]: <p>"for consecutive 5 epochs" -&gt; "for 5 consecutive epochs"</p><p> </p><p>make "Note" bold.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (AtmaMani) [inline guide/14-deep-learning/object-detection.ipynb]: <p>"Once the later layers have been sufficiently trained, the earlier layers are <code>unfreeze()</code> and fine-tuned to the" -&gt; "... the earlier layers are unfrozen (by calling <code>unfreeze()</code>) and ..."</p><p> </p><p>"Please note that this step is only optional." -&gt; remove the word 'only'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (AtmaMani) [inline guide/14-deep-learning/object-detection.ipynb]: <p>Can you add some examples here? Perhaps from the pavement cracks or some other sample?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (AngadKundra) [reply guide/14-deep-learning/object-detection.ipynb]: Could you help me with what exactly to put? The link to the notebook or some part from the notebook?

Comment (AtmaMani) [reply guide/14-deep-learning/object-detection.ipynb]: <p>Since the text is a bit abstract, you can add a part of the notebook (graphics, code snippets that show how predict video function is called) and hyperlink to the notebook for full workflow.</p><p> </p>

Comment (AngadKundra) [reply guide/14-deep-learning/object-detection.ipynb]: Would the hyperlink suffice? Since it would contain the code and the full workflow. Otherwise it would be repetition of the same thing.

**Comment by AtmaMani** [comment]
@AngadKundra looks like the object-detection file has changed when I merged the previous PR. Could you please resolve the merge conflicts and update this PR?

**Comment by AtmaMani** [comment]
@rohitgeo I agree with your suggestion, perhaps we could work on a separate PR to break out object dection on video and make that a separate page. I am inclined to merging this PR as it address many more smaller items.
